### DISCUSSION TITLE
Access files of a github runner?

### DISCUSSION THREAD

**Discussion by ktiniatros** [actions/virtual-environments#4191] [Ideas]
Hello,

one job of my github actions checked out out my repository and created locally a new commit hash. I need to retrieve that commit hash for legal reasons. Is it possible to somehow access all the files of the gitlab runner executed that job? Or download the git metadata of this job somehow?

**Comment by catthehacker** [comment]
You could commit back to repository to preserve the commit if you wish to retain it. You can also use https://github.com/actions/upload-artifact to upload a tarball of the repository, which then you can download.
